src/inmanta/server/services/orchestrationservice.py

```diff
@@ -814,9 +818,74 @@ async def _put_version(
             all_agents: set[str] = {res.agent for res in rid_to_resource.values()}
             all_agents.add(const.AGENT_SCHEDULER_ID)
 
+            type_to_agent: dict[str, str] = {res.resource_type: res.agent for res in rid_to_resource.values()}
+            LOGGER.debug(module_version_info)
+
             for agent in all_agents:
                 await self.agentmanager_service.ensure_agent_registered(env, agent, connection=connection)
 
+            async def populate_join_table(
+                cm_version: int,
+                environment: uuid.UUID,
+                module_version_info: dict[str, str] | None,
+                type_to_module_data: dict[ResourceIdStr, str] | None,
+                type_to_agent: dict[str, str] | None,
+                connection: Connection,
+            ) -> None:
+                if not all([module_version_info, type_to_module_data, rid_to_resource, type_to_agent]):
+                    LOGGER.debug(
+                        "Cannot populate join table modules_for_agent. The following arguments are not set: %s"
+                        % ", ".join(
+                            [
+                                arg_name
+                                for arg_name, arg in zip(
+                                    ["module_version_info", "type_to_module_data", "rid_to_resource", "type_to_agent"],
+                                    [module_version_info, type_to_module_data, rid_to_resource, type_to_agent],
+                                )
+                                if not arg
+                            ]
+                        )
+                    )
+                    return
+
+                assert module_version_info
+                assert type_to_module_data
+                assert rid_to_resource
+                assert type_to_agent
```

Comment:
Why is this not handled with stricter typing?

Reply:
Do you mean with non-optional types ? This is something I have been wondering about for the two new arguments to `put_version` i.e. `module_version_info` and `type_to_module_data` should we set those to non optional as well and update all the tests accordingly ?

Reply:
Or simply not call the method if the information is not present, that at least keeps this method clean